src/backend/utils/inactive-blog-filter.js

```diff
@@ -98,6 +98,8 @@ async function update(
         // We convert the dateDiff(result) from ms in days
         const timeDiff = Math.ceil(dateDiff(recentPostDate) / (1000 * 3600 * 24));
 
+        // BLOG_INACTIVE_TIME is supplied with a necessary default value, e.g. of 365 days
+        // See: https://github.com/Seneca-CDOT/telescope/issues/396
         if (timeDiff > (process.env.BLOG_INACTIVE_TIME || 365)) {
```

Comment:
Good job with the comment 👍 .

But I just realized something:

Lets say `process.env.BLOG_INACTIVE_TIME` value is greater than `365` day (Lets say 500 days). 
If a blog has only been active for 400 days it fails the first check but passes on `365` which is then flagged which I assume is what we don't want.

Maybe something like this would be better? To use only one value instead of checking both?:
```
let inactive_time = process.env.BLOG_INACTIVE_TIME != null? process.env.BLOG_INACTIVE_TIME : 365;
if (timeDiff > inactive_time) {
```

Reply:
Yup, the code actually does this! `(a || b)` is actually functionally equivalent to `(a ? a : b)` and is called a [short-circuit evaluation.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators#Short-circuit_evaluation)!

Reply:
The expression is this:
timeDiff > (a || b)

Let say the values are:
a = 500 days (process.env.BLOG_INACTIVE_TIME);
b = 365;
timeDiff = 400;

**First step**: 400 > 500 is false but since || is OR it doesn't just stop it moves onto the next expression
**Second step**: 400 > 365 which is true. 

> (a || b) is functionally equivalent to (a ? a : b)

They are the same but in this context, we are setting a default value (We don't check both 500 and 365):
```
let inactive_time = process.env.BLOG_INACTIVE_TIME != null? process.env.BLOG_INACTIVE_TIME : 365;
if (timeDiff > inactive_time) {
```

Reply:
> 
> 
> The expression is this:
> `timeDiff > (a || b)`
> 

>
> **First step**: 400 > 500 is false but since || is OR it doesn't just stop it moves onto the next expression
> **Second step**: 400 > 365 which is true.

A good thought, but, in this case, `(a || b)` is resolved *before* the binary greater-than comparison with `timeDiff` due to the enclosure of the [short-circuit evaluation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators#Short-circuit_evaluation) by a [grouping operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Grouping) (i.e. parentheses), which provides that expression with [a higher priority](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#Table).

As such, the logic in question proceeds as follows:
- If `process.env.BLOG_INACTIVE_TIME` is [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) (e.g. undefined, null, etc.), `timeDiff > (process.env.BLOG_INACTIVE_TIME || 365)` evaluates to `timeDiff > 365`. 
- If `process.env.BLOG_INACTIVE_TIME` is [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) (e.g. is defined by a `.env` file), `timeDiff < (process.env.BLOG_INACTIVE_TIME || 365)` evaluates to `timeDiff > process.env.BLOG_INACTIVE_TIME `. 

(On a side note, our professor, @humphd, actually [specifically recommended use of short-circuit evaluation](https://seneca-open-source.slack.com/archives/C8WKYE7E3/p1575216101086600) in these scenarios.)

Thank you for taking the time review my pull request in such detail, I really appreciate it!

Reply:
Oh OK. I think I get it. Thanks for providing the information. I don't have any other problems with this. Need someone else to review this code now 👍 .